src/domain/health/health.repository.ts

```diff
@@ -1,26 +1,33 @@
-import { Inject, Injectable } from '@nestjs/common';
+import { IQueueConsumerService } from '@/datasources/queues/queue-consumer.service.interface';
 import { HealthEntity } from '@/domain/health/entities/health.entity';
 import { IHealthRepository } from '@/domain/health/health.repository.interface';
 import {
   CacheReadiness,
   ICacheReadiness,
 } from '@/domain/interfaces/cache-readiness.interface';
 import { ILoggingService, LoggingService } from '@/logging/logging.interface';
-
+import { Inject, Injectable } from '@nestjs/common';
 @Injectable()
 export class HealthRepository implements IHealthRepository {
   constructor(
-    @Inject(CacheReadiness) private readonly cacheService: ICacheReadiness,
-    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @Inject(CacheReadiness)
+    private readonly cacheService: ICacheReadiness,
+    @Inject(LoggingService)
+    private readonly loggingService: ILoggingService,
+    @Inject(IQueueConsumerService)
+    private readonly queueConsumerService: IQueueConsumerService,
   ) {}
 
   async isReady(): Promise<HealthEntity> {
     try {
       await this.cacheService.ping();
-      return HealthEntity.READY;
     } catch (error) {
       this.loggingService.warn('Cache service connection is not established');
       return HealthEntity.NOT_READY;
     }
+
+    return this.queueConsumerService.isReady()
+      ? HealthEntity.READY
+      : HealthEntity.NOT_READY;
```

Comment:
By having this, `QueueConsumerService` is now a dependency while checking the service health. i.e.: `/health/ready` endpoint now depends on the status of the queue consumer. I thought the events consumption could be seen as a hard dependency for the CGW (when the FF is active), so I opted for including it here.